src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -155,6 +155,9 @@ def verify(self, signature, data, signature_algorithm):
     @classmethod
     def from_encoded_point(cls, curve, data):
         utils._check_bytes("data", data)
+        if len(data) == 0:
+            raise ValueError("data must not be an empty byte string")
```

Comment:
Nit, but can you put this check after the `curve` one?

Reply:
I can move it, but why? It seems more logical to group "is this bytes" and "is it non-empty" together to me.

Reply:
Because if you pass a non-curve and empty bytes, I want you to get the type error.